src/components/SearchBar/index.tsx

```diff
@@ -0,0 +1,28 @@
+import * as Styled from './styled';
+import { SearchIcon } from './components';
+import React, { ChangeEvent } from 'react';
+
+interface Props {
+  label: string;
+  inputValue: string;
+  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+}
+
+export class SearchBar extends React.Component<Props, unknown> {
+  constructor(props: Props) {
+    super(props);
+  }
+
```

Comment:
Good that you read this article. =) Yeah it is a point if view. 
Now we are trainig to use class based components so we are using them everywhere, but in real world even when class approach were the only approach that exists - components were devided on **stateful** and **stateless**. Stateful (class) components allowed you to declare state inside. And for stateless components - functional components were used (we couldn't declare lifecycle methods there).
And in stateful components yes, it would be safe to stick Dan advice =)